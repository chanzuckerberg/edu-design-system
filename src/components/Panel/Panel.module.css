@import '../../design-tokens/mixins.css';

/*------------------------------------*\
    #PANEL
\*------------------------------------*/

/**
 * 1) A container for content
 */
.panel {
  background: var(--eds-theme-color-background-neutral-default);
  border-radius: var(--eds-border-radius-xl);
  padding: var(--eds-size-3);
}

.panel--squared {
  padding-left: 0;
  padding-right: 0;
  border-radius: 0;
  border: none;
  box-shadow: var(--eds-box-shadow-md);
}

.panel--flush {
  padding: 0;
}

.panel--align-center {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}
