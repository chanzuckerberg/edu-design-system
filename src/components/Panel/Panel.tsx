import clsx from 'clsx';
import React, { ReactNode } from 'react';
import styles from './Panel.module.css';

export interface Props {
  /**
   * Alignment variations for the panel
   * - **center** horizontally and vertically aligns the content
   */
  align?: 'center';
  /**
   * Child node(s) that can be nested inside component
   */
  children?: ReactNode;
  /**
   * CSS class names that can be appended to the component.
   */
  className?: string;
  /**
   * Style variants
   * - **flush** removes padding from the panel
   */
  flush?: boolean;
  /**
   * Style variants
   * - **squared** squares corners, removes borders, and adds box shadow for headlines
   */
  variant?: 'squared';
}

/**
 * BETA: This component is still a work in progress and is subject to change.
 *
 * ```ts
 * import {Panel} from "@chanzuckerberg/eds";
 * ```
 *
 * TODO: update this comment with a description of the component.
 */
export const Panel = ({
  className,
  children,
  align,
  variant,
  flush,
  ...other
}: Props) => {
  const componentClassName = clsx(
    styles['panel'],
    className,
    align === 'center' && styles['panel--align-center'],
    variant === 'squared' && styles['panel--squared'],
    flush && styles['panel--flush'],
  );

  return (
    <div className={componentClassName} {...other}>
      {children}
    </div>
  );
};
