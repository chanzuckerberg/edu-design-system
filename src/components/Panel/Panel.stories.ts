import { BADGE } from '@geometricpanda/storybook-addon-badges';
import type { StoryObj, Meta } from '@storybook/react';
import type { ComponentProps } from 'react';

import { Panel } from './Panel';

export default {
  title: 'Components/Panel',
  component: Panel,
  parameters: {
    badges: ['1.0', BADGE.DEPRECATED],
  },
  args: {
    children: 'A Panel is a generic bordered container for content.',
  },
} as Meta<Args>;

type Args = ComponentProps<typeof Panel>;

export const Default: StoryObj<Args> = {};

export const Flush: StoryObj<Args> = {
  args: {
    flush: true,
  },
};

export const Squared: StoryObj<Args> = {
  args: {
    variant: 'squared',
  },
};

export const Centered: StoryObj<Args> = {
  args: {
    align: 'center',
  },
};
