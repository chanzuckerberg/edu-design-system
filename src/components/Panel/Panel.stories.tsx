import {StoryObj, Meta} from '@storybook/react';
import React from 'react';

import {Panel} from './Panel';

export default {
  title: 'Molecules/Layout and Containers/Panel',
  component: Panel,
  args: {
    children: 'A Panel is a generic bordered container for content.',
  },
} as Meta<Args>;

type Args = React.ComponentProps<typeof Panel>;

export const Default: StoryObj<Args> = {};

export const Flush: StoryObj<Args> = {
  args: {
    flush: true,
  },
};

export const Squared: StoryObj<Args> = {
  args: {
    variant: 'squared',
  },
};

export const Centered: StoryObj<Args> = {
  args: {
    align: 'center',
  },
};
