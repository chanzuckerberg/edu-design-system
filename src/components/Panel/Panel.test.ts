import {generateSnapshots} from '@chanzuckerberg/story-utils';
import * as stories from './Panel.stories';

describe('<Panel />', () => {
  generateSnapshots(stories);
});
