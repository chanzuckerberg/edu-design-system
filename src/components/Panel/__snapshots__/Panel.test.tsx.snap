// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Panel /> Centered story renders snapshot 1`] = `
<div
  class="panel panel--align-center"
>
  A Panel is a generic bordered container for content.
</div>
`;

exports[`<Panel /> Default story renders snapshot 1`] = `
<div
  class="panel"
>
  A Panel is a generic bordered container for content.
</div>
`;

exports[`<Panel /> Flush story renders snapshot 1`] = `
<div
  class="panel panel--flush"
>
  A Panel is a generic bordered container for content.
</div>
`;

exports[`<Panel /> Squared story renders snapshot 1`] = `
<div
  class="panel panel--squared"
>
  A Panel is a generic bordered container for content.
</div>
`;
